File: impls/pm-rust/step0_repl.rs
use std::io;

fn main() {
    loop {
        println!("user> ");
        let input = read();
        println!("{}", input);
    }
}

fn read() -> String {
    let mut input: String = "".to_string();
    io::stdin().read_line(&mut input)
        .expect("Failed to read line");
    input
}

File: impls/pm-js/step0_repl.js
const readline = require('readline');

function rep(input) {
    output = READ(input);
    output = EVAL(output);
    output = PRINT(output);
    return output;
}

function READ(input) {
    return input;
}

function EVAL(input) {
    return input;
}

function PRINT(input) {
    return input;
}

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    prompt: 'user> '
});

rl.prompt();

rl.on('line', (line) => {
    console.log(line);
    rl.prompt();
}).on('close', () => {
    process.exit(0);
});
